include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -37,15 +37,37 @@ class CSummedAreaTableImageFilterBase
 					const auto regions = inputImage->getRegions();
 
 					size_t retval = {};
-					for (const auto* region = regions.begin(); 
-						region < regions.end(); 
-						++region)
-						retval += region->imageExtent.width * region->imageExtent.height * region->imageExtent.depth * inputImage->getCreationParameters().arrayLayers;
+					for (const auto* region = regions.begin(); region != regions.end(); ++region)
+						retval += (region->bufferRowLength ? region->bufferRowLength : region->imageExtent.width) * region->imageExtent.height * region->imageExtent.depth;
 
-					retval *= channels * decodeByteSize;
+					retval *= inputImage->getCreationParameters().arrayLayers * channels * decodeByteSize;
```

Comment:
you're gonna do one layer at a time, and not sum across layers, so the number of layers shall not influence the scratch size.

Reply:
oh, right. logic mistake

Reply:
wait what? we wanted to decode the entire image with all the data first, then using offset perform the sum (going through the layers as well and summing the single ones, but having the entire decode image due to overlapping that could appear)

Reply:
otherwise I will be forced to do what I did before - decoding in the lambda

Reply:
it's because I use the scratch to read from it decoded values and I have to have those layers as well, cuz the basic filter sends the layer to the lambda as well. However before I had an useless approach - decoding image and then decoding extraordinary an pixel I'm operating on (decoding 2 times) so I find my way good now

Reply:
solved

Reply:
> but having the entire decoded image due to overlapping that could appear)
You can run the decode,sum,encode routines not once for all layers, but multiple times, one layer at a time.
> cuz the basic filter sends the layer to the lambda too. 
copy the state, set the layer count to 1, adjust the base layers for input and output when performing each loop